### PR TITLE
Extra control placement 3039

### PR DESCRIPTION
**Pull request by Ethan-Guttman**
This closes the good first issue #3039 (Allow controls to be placed at top-center, bottom-center, left-center, and right-center)

example pictures showcasing all four new positions:
image
image

Multiple ctrls will have margins between themselves like the already existing positions. However, depending on their widths in the top and bottom center they can look weird (smaller elements floating left).

I ordered the positions code in the css and map.js files by going from top-left to center-left clockwise listing each position too.

### CODE REVIEW

**Comment by mourner** [comment]
Sorry for the late response — this PR is on our mind and we'll look at it closer after we're done with the current v3.3 release. 🙏

**Comment by Ethan-Guttman** [comment]
No problem, thanks for the update!

File: src/ui/map.js
Comment (rreusser) [inline src/ui/map.js:86]: It seems like `left-center` and `right-center` are flipped relative to the others. Would putting the vertical positioning consistently first be easier to remember?

PR suggestion:
| | | |
|-|-|-|
| top-left | top-center | top-right |
| **left-center** | | **right-center** |
| bottom-left | bottom-center | bottom-right |

Maybe easier to remember:
| | | |
|-|-|-|
| top-left | top-center | top-right |
| **center-left** | | **center-right** |
| bottom-left | bottom-center | bottom-right |

Comment (Ethan-Guttman) [reply src/ui/map.js:86]: It's a bit of a tossup because I wanted to avoid having "center" show up as both the first and last word. Maybe make "bottom-center" into "center-bottom" too (as well as with "top-center")? Or maybe make those two "bottom-middle" and "top-middle"?

| Suggestion 1 |  | |
| ------------- | ------------- | -----------|
| top-left  | **center-top**  | top-right |
| **center-left**  |                            | **center-right** |
| bottom-left  | **center-bottom**  | bottom-right |

| Suggestion 2 |  | |
| ------------- | ------------- | -----------|
| top-left  | **top-middle** | top-right |
| **center-left**  |                            | **center-right** |
| bottom-left  | **bottom-middle**  | bottom-right |

I don't know which is best, seems like there are only tradeoffs unless we bite the bullet with my suggestion 2 and have "middle" and "center" refer to different axes

Comment (rreusser) [reply src/ui/map.js:86]: That makes sense. I know I'm engaging in some drive-by nitpicking; it just reminds me of properties I have to look up every single time, e.g. flexbox alignment vs. justification and SVG alignment-baseline center vs. central vs. middle. So I don't have a strong stance, just that some amount of logic, precedent, or memorability would be a nice goal.

Comment (rreusser) [reply src/ui/map.js:86]: I'm not sure I like this, but what about:

| Suggestion 3 |  | |
| ------------- | ------------- | -----------|
| top-left  | **top** | top-right |
| **left**  |                            | **right** |
| bottom-left  | **bottom**  | bottom-right |

Comment (Ethan-Guttman) [reply src/ui/map.js:86]: That could work and I'm leaning towards it the most. Thanks for this suggestion

**Comment by rizvanovic** [comment]
This would be sick to get out

**Comment by JabSYsEmb** [comment]
It seems that no changes since last comment by @Ethan-Guttman.
I can take over the issue if you're tight on time... let me know

**Comment by Ethan-Guttman** [comment]
> It seems that no changes since last comment by @Ethan-Guttman. I can take over the issue if you're tight on time... let me know

I was waiting for a contact from the Mapbox team before updating again (since they said they would look at it closer later). I'll make the changes myself

**Comment by stepankuzmin** [comment]
Hey @Ethan-Guttman,

Thanks for the contribution, and I'm sorry for the delay. Is there a chance you could rebase your changes on top of the latest `main`? Overall, this looks good to me, so resolving the conflicts seems to be the last thing we need to do before landing this.

**Comment by Ethan-Guttman** [comment]
@stepankuzmin 
I'll try to rebase it by end of day today. Thanks for the go ahead!

**Comment by Ethan-Guttman** [comment]
I'm at a bit of a loss here. This is my first ticket and I have clicked the link to sign the CLA multiple times. It might just not be reflected here as it does sometimes indicate that I have signed it when I click the link. Aside from that if there is something genuinely wrong with the mac safari version I can't test it myself (having no apple devices). Those are the only failing checks

**Comment by stepankuzmin** [comment]
Hey @Ethan-Guttman,

The rebase seems wrong because your changes were made before we transitioned to TypeScript, and now your PR has 2k files changed. It'd be easier for maintainers to update your PR directly. Could you please make sure that you've enabled [edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) on that PR?

**Comment by Ethan-Guttman** [comment]
@stepankuzmin 
image
Thank you for the help and let me know if anything else is needed

**Review by stepankuzmin — APPROVED** [review]

**Comment by stepankuzmin** [comment]
Thanks for the contribution, @Ethan-Guttman, merging 👍

**Pull request merged**
